### PR TITLE
Files docs

### PR DESCRIPTION
**Pull request by ntarocco**

### CODE REVIEW

File: docs/tutorials/handling-files.rst [hunk 1]
```diff
@@ -0,0 +1,591 @@
...
+one ``default`` Location.
+
+For example, you can define a new default location named ``shared`` in the path
+``/mnt/shared``:
+
+.. code-block:: console
+
+    $ pipenv run invenio files location shared /mnt/shared --default
+
+From now on, any new Bucket (not the existing ones) will use this location
```
Comment (Glignos) [inline docs/tutorials/handling-files.rst]:
```suggestion
From now on, any new Bucket, besides the existing ones, will use this location
```

File: docs/tutorials/handling-files.rst [hunk 2]
```diff
@@ -0,0 +1,591 @@
...
+and therefor files will be stored in ``/mnt/shared``.
```
Comment (Glignos) [inline docs/tutorials/handling-files.rst]:
```suggestion
and therefore files will be stored in ``/mnt/shared``.
```

File: docs/tutorials/handling-files.rst [hunk 3]
```diff
@@ -0,0 +1,591 @@
...
+
+Integration with records
+------------------------
+Invenio integrates files and records by creating a reference
+between a Record and a Bucket. Usually a record has a reference to one
+bucket and a bucket to one record.
+
+.. image:: ../_static/invenio-files-integration.png
+
+Invenio allows you to set up different scenarios and have, for example,
```
Comment (Glignos) [inline docs/tutorials/handling-files.rst]: Maybe we need to specify that this behavior does not come out of the box and that the user needs to implement their own code to and override parts of the existing class to achieve this.

File: docs/tutorials/handling-files.rst [hunk 4]
```diff
@@ -0,0 +1,591 @@
...
+You can read more in the
+`Invenio-Files-REST <http://invenio-files-rest.readthedocs.io/en/latest/>`__
+module documentation.
+
+.. _handling-files-integration-with-records:
+
+Integration with records
+------------------------
+Invenio integrates files and records by creating a reference
+between a Record and a Bucket. Usually a record has a reference to one
```
Comment (Glignos) [inline docs/tutorials/handling-files.rst]:
```suggestion
between a Record and a Bucket. By default in Invenio a record has a reference to one
```

File: docs/tutorials/handling-files.rst [hunk 5]
```diff
@@ -0,0 +1,591 @@
...
+In a few words, you can think to Location, Storage and FileInstance as
+the physical representation of your files system(s).
+
+Object storage
+++++++++++++++
+Invenio provides an abstraction of your physical file system with an object
+storage representation similar to Amazon S3. This allows great flexibility
+and easy to use APIs.
+
+With an analogy to the traditional file systems where files are
```
Comment (Glignos) [inline docs/tutorials/handling-files.rst]:
```suggestion
You can compare this implementation with a traditional file system where files are
```

File: docs/tutorials/handling-files.rst [hunk 6]
```diff
@@ -0,0 +1,591 @@
...
+contained in folders, in Invenio object storage ``ObjectVersions``
```
Comment (Glignos) [inline docs/tutorials/handling-files.rst]:
```suggestion
contained in folders, in Invenio object storage the ``ObjectVersions``
```

File: docs/tutorials/handling-files.rst [hunk 7]
```diff
@@ -0,0 +1,591 @@
...
+are contained in ``Buckets``.
```
Comment (Glignos) [inline docs/tutorials/handling-files.rst]:
```suggestion
are contained inside the ``Buckets``.
```

File: docs/tutorials/handling-files.rst [hunk 8]
```diff
@@ -0,0 +1,591 @@
...
+
+Buckets are a set of objects, they are uniquely identified by an ID
```
Comment (Glignos) [inline docs/tutorials/handling-files.rst]:
```suggestion
Buckets are sets of objects, they are uniquely identified by an ID
```

File: docs/tutorials/handling-files.rst [hunk 9]
```diff
@@ -0,0 +1,591 @@
...
+and can define constraints on file sizes or quotas. A bucket also
+defines the default Location to use when adding files, if none provided.
```
Comment (Glignos) [inline docs/tutorials/handling-files.rst]:
```suggestion
defines the default Location to use when adding files, if none is provided.
```

File: docs/tutorials/handling-files.rst [hunk 10]
```diff
@@ -0,0 +1,591 @@
...
+
+Overview
+--------
+The first step when setting up files integration is to decide where to store them.
+In Invenio, you can achieve this by defining a ``Location``: a location
+is a representation of your storage system. It can be, for example, a local
+folder in your machine or a remote storage. It has a ``name`` and a ``URI`` to
+define the base path to the files.
+
+To access and manage files in a location, Invenio uses ``Storage`` implementation:
```
[the remark this answers is not in the document]
Comment (ntarocco) [reply docs/tutorials/handling-files.rst]: Actually I put here `a Storage implementation`, it makes more sense IMHO.

File: docs/tutorials/handling-files.rst [hunk 11]
```diff
@@ -0,0 +1,591 @@
...
+it defines how to access files in your location. In other words, it knows how files
+are stored, e.g. in a hierarchy of folders, and how to read and write them.
+
+You can define multiple locations and storage. This is useful, for example,
```
[the remark this answers is not in the document]
Comment (ntarocco) [reply docs/tutorials/handling-files.rst]: Actually, when writing the doc the first time, I also used `storages`, but it is actually uncountable, no plural!

File: docs/tutorials/handling-files.rst (hunk 3 above)
Comment (ntarocco) [reply docs/tutorials/handling-files.rst]: Right, I have added `To achieve any of these, you will have to define
your own integration between records and files.`

**Comment by ntarocco** [comment]
@Glignos I have applied all your suggestions, thanks a lot! There was nothing after line 200 something, I hope this was intended and there was no issue with comments!

**Comment by Glignos** [comment]
> @Glignos I have applied all your suggestions, thanks a lot! There was nothing after line 200 something, I hope this was intended and there was no issue with comments!

@ntarocco It is intended indeed :)

**Pull request merged**